getting-started/data-node-authentication.md

```diff
@@ -11,90 +11,181 @@ and the access node need to have authentication set up for those
-default name of the HBA file is `pg_hba.conf`. Details of [the
-`pg_hba.conf` format can be found in the PostgreSQL
+In order to set up password authentication for `new_user` using
+password `xyzzy`, you need to:
+
+1. Set the correct password encryption in the PostgreSQL configuration
+   file.
+2. Check that the correct HBA (Host-Based Authentication) file is in
+   the PostgreSQL configuration file.
+3. Update the HBA file on the data nodes to enable password
+   authentication.
+4. Reload the server to read the new configuration.
+5. Add the user on the data node with the right password.
+
+### Set up password encryption method
+
+First, you need to edit the PostgreSQL configuration file
+`postgresql.conf` in the data directory and set `password_encryption`
+to the right password encryption method, in our case
+`scram-sha-256`. The configuration file should contain the following
+lines:
+
+```
+password_encryption = 'scram-sha-256'		# md5 or scram-sha-256
+```
+
+>:TIP: A quick way to edit the configuration file is by using Perl:
+>
+> ```
+> perl -pi -e "s/^#?(password_encryption) *= *\S+/\1 = 'scram-sha-256'/" \
+>    postgresql.conf
+> ```
+>
+> This will rename the original configuration file to
+> `postgresql.conf.orig` and set `password_encryption` in
+> `postgresql.conf`.
+
+### Enable password authentication
+
+The default name of the HBA file is `pg_hba.conf`, so check the
+configuration file (normally `postgresql.conf` in the data directory
+of the data node server) and make sure that the `hba_file` option is
+commented out. That part of the file should look like this:
+
+```
+#hba_file = 'ConfigDir/pg_hba.conf' # host-based authentication file
+                                    # (change requires restart)
+```
+
+After that, you should add a line to the `pg_hba.conf` file giving
+`new_user` in the `example.com` domain access to all databases on the
+data node and is identified using SCRAM SHA-256.
+
+```
+# IPv4 local connections:
+host    all             new_user        .example.com            password
+```
+
+Details of [the `pg_hba.conf` format can be found in the PostgreSQL
 manual.][postgresql-hba].
 
-In this case, you want to give `new_user` in the `example.com` domain
-access to all databases on the data node, so add the following line to
-the `pg_hba.conf`.
+### Reload server configuration
+
+To reload the server configuration, you can use the following command
+on the data node:
 
 ```
-host    all    new_user    .example.com    scram-sha-256
+pg_ctl reload
 ```
 
-Once you have set up the authentication method on the data node, you
-need to check the configuration file (normally `postgresql.conf` in
-the data directory of the data node server) and make sure that the
-`hba_file` option is correctly set.
+### Add new user on data node
+
+You can add the new user on the data node by connecting to it using
+`psql` and using [`CREATE ROLE`][postgresql-create-role]:
+
+```
```

Comment:
I notice most of the other documentation uses a ` ```sql` header to start a sql command block.  I'm also now noticing there's a ` ```bash` header as well, which is something I haven't been using either.  I'm not sure if either of these has any practical impact.

Reply:
I think they can use keyword highlighting for it. Added.